### DISCUSSION TITLE
Preferred way to test new builds

### DISCUSSION THREAD

**Discussion by luisfcorreia** [probonopd/MiniDexed#112] [Q&A] [answered]
so, I've got some changes I want to test them

running ./build.sh produces (amongst other files) a "kernel7.img" (for a raspi 2)

is it enough to just place that file in the SD card?

**Comment by probonopd** [comment, marked as answer]
Yes.

**Comment by luisfcorreia** [comment]
Related, how is debugging done?
I've seen some notes about printing to midi out, is that "it"?

(I would like to be able to test this in my Linux desktop :) )

**Reply by probonopd** [reply]
Some diagnostic/debug messages are printed out to HDMI. Is that you are looking for?

**Reply by luisfcorreia** [reply]
I'm usually "team printf()" :)
would that go to hdmi screen?

**Reply by probonopd** [reply]
Teammate! Here we use `mLogger.Write()`

https://github.com/probonopd/MiniDexed/blob/8be0781b492149c5bbfa507e1f1d17d1a19b81ed/src/circle_stdlib_app.h#L196-L197

and `printf`

https://github.com/probonopd/MiniDexed/blob/d6025f194319fea7a205621a2aa81bd29ca0fd2f/src/mididevice.cpp#L86

and both show up on the HDMI screen, although I think `mLogger.Write()` is "the way" to do this.

**Reply by luisfcorreia** [reply]
I can confirm, hdmi debugging works as expected!

thanks
